geomstats/geometry/klein_bottle.py

```diff
@@ -0,0 +1,392 @@
+        """Find the representative of point2 which is closest to point1.
+
+        Only representatives in the 8 surrounding
+        squares have to be considered.
+
+        Parameters
+        ----------
+        point1: array-like, shape [..., 2]
+            Point on the manifold.
+        point2: array-like, shape [..., 2]
+            Point on the manifold.
+
+        Returns
+        -------
+        point1: array-like, shape [...,2]
+            Canonical representation of point1 in unit square.
+        minimizers: array-like, shape [...,2]
+            Representation of point2 with smallest distance to point1.
+        """
+        point1, point2 = gs.broadcast_arrays(point1, point2)
+        shape = point1.shape
+        p1 = self._space.regularize(point1)
+        p1 = gs.reshape(p1, (-1, 2))
+        p2 = self._space.regularize(point2)
+        p2 = gs.reshape(p2, (-1, 2))
+        p2_2 = gs.stack([p2[:, 0], p2[:, 1] - 1], axis=-1)
+        p2_3 = gs.stack([p2[:, 0], p2[:, 1] + 1], axis=-1)
+        p2_4 = gs.stack([p2[:, 0] + 1, 1 - p2[:, 1]], axis=-1)
+        p2_5 = gs.stack([p2[:, 0] + 1, 2 - p2[:, 1]], axis=-1)
+        p2_6 = gs.stack([p2[:, 0] + 1, 0 - p2[:, 1]], axis=-1)
+        p2_7 = gs.stack([p2[:, 0] - 1, 1 - p2[:, 1]], axis=-1)
+        p2_8 = gs.stack([p2[:, 0] - 1, 2 - p2[:, 1]], axis=-1)
+        p2_9 = gs.stack([p2[:, 0] - 1, 0 - p2[:, 1]], axis=-1)
+        p2_total = gs.stack(
+            [p2, p2_2, p2_3, p2_4, p2_5, p2_6, p2_7, p2_8, p2_9], axis=0
+        )
+        indices = gs.argmin(
+            gs.sum((p2_total - gs.expand_dims(p1, 0)) ** 2, axis=-1), axis=0
+        )
+        minimizers = gs.empty_like(p1)
+        for i, index in enumerate(indices):
+            minimizers[i, :] = p2_total[index, i, :]
+        p1 = gs.reshape(p1, shape)
+        minimizers = gs.reshape(minimizers, shape)
+        return p1, minimizers
+
+
+class KleinBottle(Manifold):
+    r"""Class for the Klein Bottle, a two dimensional manifold which is not orientable.
+
+    Points are understood to be on the Klein Bottle if they are in the interval
+    :math:`[0,1]^2`.
+    Each point in :math:`\mathbb R^2` can be understood as a point on the Klein Bottle
+    by considering the equivalence relation
+
+    .. math::
+       :nowrap:
+
+       \begin{align}
+         &  (x_1,y_1) \sim (x_2, y_2) \\
+         \Leftrightarrow \quad &  y_1-y_2 \in \mathbb Z
+                                 \text{ and } x_1-x_2 \in 2\mathbb Z \\
+         \text{ or } \quad &
+        y_1 + y_2 \in \mathbb Z \text{ and } x_1-x_2 \in \mathbb Z\setminus 2\mathbb Z.
+       \end{align}
+    """
+
+    def __init__(self, equip=True):
+        dim = 2
+        super().__init__(
+            dim=dim, shape=(dim,), metric=None, default_coords_type="intrinsic"
+        )
+
+        if equip is True:
+            self.equip_with_metric()
+
+    def _default_metric(self):
+        return KleinBottleMetric
+
+    def equip_with_metric(self, Metric=None):
```

Comment:
Metric --> metric
(no uppercases in python unless it's for classes)

Reply:
We might have to remove the _default_metric and equip_with_metric methods, and define the metric directly in the init instead (for consistency for the current API)